src/config.c

```diff
@@ -274,3 +310,95 @@ int load_config(struct vpn_config *cfg, const char *filename)
 
 	return ret;
 }
+
+void destroy_vpn_config(struct vpn_config *cfg)
+{
+	free(cfg->pppd_log);
+	free(cfg->pppd_plugin);
+	free(cfg->pppd_ipparam);
+	free(cfg->pppd_ifname);
+	free(cfg->pppd_call);
+	free(cfg->ca_file);
+	free(cfg->user_cert);
+	free(cfg->user_key);
+	free(cfg->cipher_list);
+	while (cfg->cert_whitelist != NULL) {
+		struct x509_digest *tmp = cfg->cert_whitelist->next;
+		free(cfg->cert_whitelist);
+		cfg->cert_whitelist = tmp;
+	}
+}
+
+void merge_config(struct vpn_config *dst, struct vpn_config *src)
+{
+	if (src->gateway_host[0])
+		strcpy(dst->gateway_host, src->gateway_host);
+	if (src->gateway_port != invalid_cfg.gateway_port)
+		dst->gateway_port = src->gateway_port;
+	if (src->username[0])
+		strcpy(dst->username, src->username);
+	if (src->password[0])
+		strcpy(dst->password, src->password);
+	if (src->otp[0])
+		strcpy(dst->otp, src->otp);
```

Comment:
cookie is in the structure here in between, but doesn't need to be merged (it isn't obtained from cli nor from config file). Maybe we should insert a comment about this?

Reply:
I'd actually rather move cookie into the `tunnel` struct, I can't think of any good reason for it to be in the config.
Any reason not to?